src/randoop/main/GenTests.java

```diff
@@ -429,264 +399,335 @@ public boolean handle(String[] args) throws RandoopTextuiException {
-      } catch (Exception e) {
-        throw new Error ("can't create temp file", e);
+      //And, generate regression tests, unless user says not to
+      if (! GenInputsAbstract.no_regression_tests) {
+        checkTest = checkTest.or(new RegressionTestPredicate());
       }
-      write_junit_tests ("./before_clean", sequences, null);
+      outputTest = baseTest.and(checkTest);
+    }
+    return outputTest;
+  }
 
-      write_sequences (sequences, tmpfile.getPath());
-      generate_clean_checks (tmpfile.getPath());
+  /**
+   * Outputs JUnit tests for the sequence list, and if user indicates, also,
+   * writes serialized tests.
+   * 
+   * @param sequences  the sequences to output
+   * @param junitPrefix  the filename prefix for test output
+   */
+  private void outputTests(List<ExecutableSequence> sequences, String junitPrefix) {
+    if (output_tests_serialized != null) {
+      outputSerializedTests(sequences, junitPrefix);
     }
 
-    // Run the tests a second time, looking for any different checks
-    // This removes any checks whose values are not deterministic
-    // (such as values dependent on the current date/time)
-    if (GenInputsAbstract.compare_checks) {
-      write_junit_tests ("./before_cmp", sequences, null);
+    writeJUnitTests (junit_output_dir, sequences, null, junitPrefix);
+  }
+  
+  /**
+   * Outputs serialized version of test sequences using given string as a
+   * suffix for the file name.
+   * 
+   * @param sequences  the list of test sequences to serialize
+   * @param suffix  the string to use as a suffix
+   * @throws Error if an exception is thrown while writing to the file
+   */
+  private void outputSerializedTests(List<ExecutableSequence> sequences, 
+      String suffix) throws Error {
+    System.out.println("Serializing tests...");
+    try {
 
-      remove_diff_checks (tmpfile.getPath());
-      sequences = read_sequences (tmpfile.getPath());
+      FileOutputStream fileos = new FileOutputStream(output_tests_serialized + suffix);
+      ObjectOutputStream objectos = new ObjectOutputStream(new GZIPOutputStream(fileos));
+      objectos.writeObject(sequences);
+      objectos.close();
+      fileos.close();
+
+    } catch (Exception e) {
+      throw new Error(e);
     }
+  }
 
-    // Write out junit tests
-    if (GenInputsAbstract.outputlimit < sequences.size()) {
-      List<ExecutableSequence> seqs = new ArrayList<ExecutableSequence>();
-      for (int ii = 0; ii < GenInputsAbstract.outputlimit; ii++)
-        seqs.add (sequences.get (ii));
-      sequences = seqs;
+  /**
+   * Serializes component sequences used by the generator.
+   *  
+   * @param gen  the sequence generator
+   * @throws Error if an exception is thrown while writing to the file
+   */
+  private void outputComponentSequences(ForwardGenerator gen) throws Error {
+    // Output component sequences.
+    System.out.print("Serializing component sequences...");
+    try {
```

Comment:
It seems that much of this method and the previous one can be extracted into a single method that takes an object and a filename and calls writeObject into the file, handling all the streams and exception catching.